src/components/FlightDetails.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import {
+  getCurrentTime,
+  getTimeFromDateString,
+  getStatusColor,
+} from "../utility";
+import useFlightApi from "../Apis";
+
+const FlightDetails = () => {
+  const history = useNavigate();
+  const [flightDetails, setFlightDetails] = useState([]);
+  const [errorApi, setErrorApi] = useState(null); // State to store error information
+  const { id } = useParams();
+
+  const { data, error = null } = useFlightApi(
+    `https://flight-status-mock.core.travelopia.cloud/flights/${id}`
+  );
+
+  useEffect(() => {
+    setFlightDetails(data);
```

Comment:
why are you setting data again in the component state ? why not use it directly ? same for error as well